src/org/opendatakit/briefcase/ui/reused/transfer/sourcetarget/AggregateServerDialogForm.java

```diff
@@ -77,8 +91,25 @@
 
     connectButton.addActionListener(__ -> triggerConnect());
 
+    urlField.addFocusListener(new FocusAdapterBuilder()
+        .onFocusLost(e -> urlField.setText(AggregateServer.cleanUrl(urlField.getText())))
+        .build());
+
+    urlValidator = UiFieldValidator.of(urlField, urlFieldLabel, REQUIRED, URI).onChange(this::updateConnectButton);
+    usernameValidator = UiFieldValidator.of(usernameField, usernameFieldLabel, EMAIL.negate()).onChange(this::updateConnectButton);
+    passwordValidator = UiFieldValidator.of(passwordField, passwordFieldLabel).onChange(this::updateConnectButton);
+    usernameAndPasswordValidator = UiLinkedFieldValidator.of(usernameValidator, passwordValidator).onChange(this::updateConnectButton);
+
     getRootPane().setDefaultButton(connectButton);
+  }
 
+  private void updateConnectButton() {
+    connectButton.setEnabled(Stream.of(
+        urlValidator.isValid(),
+        usernameValidator.isValid(),
+        passwordValidator.isValid(),
+        usernameAndPasswordValidator.isValid()
+    ).reduce(true, Boolean::logicalAnd));
```

Comment:
Usually I love your streams, man, but this one increases complexity for no benefit that I can see.
<img width="1109" alt="Screen Shot 2019-07-28 at 12 57 19 PM" src="https://user-images.githubusercontent.com/55099/62012173-8b4e6880-b137-11e9-9dda-9a5659d0101e.png">

Reply:
I’d be all for it if it were as simple as
`Seq(urlValidator, usernameValidator, passwordValidator, usernameAndPasswordValidator).forall(_.isValid)`